address.go

```diff
@@ -75,30 +75,40 @@ func (wallet *Wallet) AddressInfo(address string) (*AddressInfo, error) {
 	return addressInfo, nil
 }
 
+// CurrentAddress gets the most recently requested payment address from the
+// wallet. If that address has already been used to receive funds, the next
+// chained address is returned.
 func (wallet *Wallet) CurrentAddress(account int32) (string, error) {
 	if wallet.IsRestored && !wallet.HasDiscoveredAccounts {
 		return "", errors.E(ErrAddressDiscoveryNotDone)
 	}
 
 	addr, err := wallet.Internal().CurrentAddress(uint32(account))
 	if err != nil {
-		log.Error(err)
+		log.Errorf("CurrentAddress error: %w", err)
 		return "", err
 	}
 	return addr.String(), nil
 }
 
+// NextAddress returns the address immediately following the last requested
+// payment address. If that address has already been used to receive funds,
+// the next chained address is returned.
 func (wallet *Wallet) NextAddress(account int32) (string, error) {
 	if wallet.IsRestored && !wallet.HasDiscoveredAccounts {
 		return "", errors.E(ErrAddressDiscoveryNotDone)
 	}
 
-	addr, err := wallet.Internal().NewExternalAddress(wallet.shutdownContext(), uint32(account), w.WithGapPolicyWrap())
+	// NewExternalAddress increments the lastReturnedAddressIndex but does
+	// not return the address at the new index. That is returned below by
+	// CurrentAddress.
+	_, err := wallet.Internal().NewExternalAddress(wallet.shutdownContext(), uint32(account), w.WithGapPolicyWrap())
```

Comment:
I decided against manually incrementing the last returned address index because of this gap policy. It would not be trivial to calculate the difference between the last used and last returned indices in order to force-wrap the returned index. I should add (in a forced commit shortly) that this is a work around until if and when upstream harmonizes `NewExternalAddress` and `CurrentAddress`.